test/hts-precompile/erc-20/ERC20Contract.js

```diff
@@ -30,6 +40,139 @@ describe("ERC20Contract tests", function () {
 
   it("should be able to get token totalSupply", async function () {
     const totalSupply = await erc20Contract.totalSupply(tokenAddress);
-    expect(totalSupply).to.equal(1000);
+    expect(totalSupply).to.equal(TOTAL_SUPPLY);
+  });
+
+  it("should be able to get token balance of any account", async function () {
+    const contractOwnerBalance = await erc20Contract.balanceOf(tokenAddress, tokenCreateContract.address);
+    const wallet1Balance = await erc20Contract.balanceOf(tokenAddress, signers[0].address);
+    const wallet2Balance = await erc20Contract.balanceOf(tokenAddress, signers[1].address);
+
+    expect(contractOwnerBalance).to.exist;
+    expect(contractOwnerBalance.toNumber()).to.eq(TOTAL_SUPPLY - INITIAL_BALANCE);
+    expect(wallet1Balance).to.exist;
+    expect(wallet1Balance.toNumber()).to.eq(INITIAL_BALANCE);
+    expect(wallet2Balance).to.exist;
+    expect(wallet2Balance.toNumber()).to.eq(0);
+  });
+
+  it("should NOT be able to use transfer", async function () {
```

Comment:
Q: Shouldn't there also be a successful example of this?

Reply:
Those HTS methods cannot be called directly from a contract, that is why we have added the `delegate` methods, although they are not part of the ERC20 standard. The same is true for this [PR](https://github.com/hashgraph/hedera-smart-contracts/pull/124)